src/libraries/System.Security.Cryptography.X509Certificates/tests/RevocationTests/AiaTests.cs

```diff
@@ -14,124 +14,128 @@ public static class AiaTests
         [Fact]
         public static void EmptyAiaResponseIsIgnored()
         {
-            CertificateAuthority.BuildPrivatePki(
-                PkiOptions.AllRevocation,
-                out RevocationResponder responder,
-                out CertificateAuthority root,
-                out CertificateAuthority[] intermediates,
-                out X509Certificate2 endEntity,
-                intermediateAuthorityCount: 2,
-                pkiOptionsInSubject: false,
-                testName: nameof(EmptyAiaResponseIsIgnored));
-
-            using (responder)
-            using (root)
-            using (CertificateAuthority intermediate1 = intermediates[0])
-            using (CertificateAuthority intermediate2 = intermediates[1])
-            using (endEntity)
-            using (ChainHolder holder = new ChainHolder())
-            using (X509Certificate2 intermediate2Cert = intermediate2.CloneIssuerCert())
-            {
-                responder.RespondEmpty = true;
-
-                X509Chain chain = holder.Chain;
-                chain.ChainPolicy.ExtraStore.Add(intermediate2Cert);
-                chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
-                chain.ChainPolicy.VerificationTime = endEntity.NotBefore.AddMinutes(1);
-                chain.ChainPolicy.UrlRetrievalTimeout = DynamicRevocationTests.s_urlRetrievalLimit;
-                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
-
-                Assert.False(chain.Build(endEntity));
-                X509ChainStatusFlags chainFlags = chain.AllStatusFlags();
-                Assert.True(chainFlags.HasFlag(X509ChainStatusFlags.PartialChain), $"expected partial chain flags, got {chainFlags}");
-                Assert.Equal(2, chain.ChainElements.Count);
-            }
+            RetryHelper.Execute(() => {
+                CertificateAuthority.BuildPrivatePki(
```

Comment:
On the one hand, not capturing closures is nice.  On the other, recreating the PKI pieces and listener and such feels like it could be hiding bugs.. so we probably only want to retry the inner portion of the usings (though moving ChainHolder inside the retry so we're not introducing a "works on the second call to Build" bug)

Reply:
Eh, I'm wary of making that even more non-deterministic based on [my experience last time][1] from the AIA tests. But those tests for sensitive to execution time, so maybe I am applying the wrong logic. I suppose it can't hurt to try the more narrow way first.

[1]: https://github.com/dotnet/runtime/pull/42680